--- public/js/head.load.js
+++ public/js/head.load.js
@@ -496,7 +496,9 @@
         ele.defer = false;
 
         // timout for asset loading
-        asset.errorTimeout = undefined;
+        asset.errorTimeout = win.setTimeout(function () {
+            error({ type: "timeout" });
+        }, 7e3);
 
         // use insertBefore to keep IE from throwing Operation Aborted (thx Bryan Forbes!)
         var head = doc.head || doc.getElementsByTagName("head")[0];
